azerothcore/azerothcore-wotlk
# fix(Scripts/MoltenCore): Fixed shazzrah "23138 - Gate of Shazzrah" teleport

23138 - Gate of Shazzrah should teleport random player which is not in his melee range, reset his threat and force him to attack teleported player - this PR exact addresses that and fixes some code mistakes and corrects this spell behavior


## Changes Proposed:
-  Fixes "23138 - Gate of Shazzrah" spell cast behavior

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested with two players, i placed player in longer distance and checked how spell handles teleport. To debug, i used some extra printfs actually (printfs are not present in this PR)


## How to Test the Changes:

**### You must use least two characters to test this PR!**, one character should engage boss and another should work as dummy. You should use ".cheat god on" or least some immune aura which doesn;t prevents agroo state.
1. Place one character in longer distance, can be 10 yards, 15 or 30 yards, whenever you like (just should be visible distance to boss)
2. With another character engage boss
3. During fight observe teleport behavior

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
